esphomeyaml/cookbook/sonoff-s2x-smartswitch.rst

```diff
@@ -0,0 +1,258 @@
+    :align: center
+    :width: 30.0%
+
+This is a basic firmware for the Sonoff S2X smart switch series. The features are:
+ 
+* The switch is automatically registered to HA via MQTT.
+* A *single click* toggles the relay (works offline).
+* A *double click*, a *triple click*, or a *longpress* triggers custom actions on HA. 
+
+.. warning::
+
+    Do NOT connect your device to the mains when programming it.
+    Take care working with the mains voltage at all times!
+
+1. Preparation
+--------------
+Prepare your socket according to these instructions 
+
+- :doc:`Sonoff S20 </esphomeyaml/devices/sonoff_s20>`
+- `Sonoff S26 <https://github.com/arendst/Sonoff-Tasmota/wiki/Sonoff-S26-Smart-Socket/>`
+
+2. ESPHome Configuration
+------------------------
+
+Here is the configuration with the basic operations outlined above.
+
+.. code-block:: yaml
+    substitutions:
+      devicename: sonoff1
+
+    esphomeyaml:
+      name: $devicename
+      platform: ESP8266
+      board: esp01_1m
+      board_flash_mode: dout
+
+
+    wifi:
+      networks:
+        - ssid: !secret wifi_ssid
+          password: !secret wifi_password
+        - ssid: !secret wifi_ssid2
+          password: !secret wifi_password2
+    # power_save_mode: none
+      ap:
+        ssid: "$devicename AP"
+        password: !secret  ap_password 
+        channel: 10
+
+    mqtt:
+      id: mqtt_client
+      broker: !secret mqtt_broker
+      username: !secret mqtt_user 
+      password: !secret mqtt_password
+
+    logger:
+      level: debug
+
+    debug:
+
+    ota:
+
+    binary_sensor:
+      - platform: gpio
+        pin:
+          number: GPIO0
+          mode: INPUT_PULLUP
+          inverted: True
+        name: "Button"
+        internal: True
+
+        on_press:
+          then:
+            - light.turn_on:
+                id: green_led_light
+                transition_length: 0s
+        
+        on_release:
+          then:
+            - light.turn_off:
+                id: green_led_light
+                transition_length: 0s
+
+        on_multi_click:
+        - timing: # Double Clicked
+            - ON for at most 1s
+            - OFF for at most 1s
+            - ON for at most 1s
+            - OFF for at least 0.2s
+          then:
+            - logger.log: "Double Clicked"
+            - lambda: |-  
+                id(double_click).publish_state(true);
+            - light.turn_on:
+                id: green_led_light
+                transition_length: 0.5s
+            - delay: 0.5s
+            - light.turn_off:
+                id: green_led_light
+                transition_length: 0.5s
+            - delay: 0.5s
+            - light.turn_on:
+                id: green_led_light
+                transition_length: 0.5s
+            - delay: 0.5s
+            - light.turn_off:
+                id: green_led_light
+                transition_length: 0.5s
+            - lambda: |-  
+                id(double_click).publish_state(false);
```

Comment:
Please add a comment (or two) as for what this block does. It's quite hard to parse what these `then` blocks are supposed to do.